### PR TITLE
Extended getVal method in Model class to accept Expr instances.

### PR DESCRIPTION
**Pull request by mcitoler**
Now you can evaluate a polynomial expression after optimizing a model. This covers part of https://github.com/SCIP-Interfaces/PySCIPOpt/issues/280. We are till missing evaluation of GenExpr.

### CODE REVIEW

File: src/pyscipopt/expr.pxi [hunk 1]
```diff
@@ -109,6 +109,13 @@ class Term:
     def __repr__(self):
         return 'Term(%s)' % ', '.join([str(v) for v in self.vartuple])
 
+    def _evaluate(self, solution):
```
Comment (mattmilten) [inline src/pyscipopt/expr.pxi]: Can you rename `solution` to `value`?

This also applies to later occurrences in this file. In my opinion there is no notion of a `solution` within the context of `Term` and `Expr`.

File: src/pyscipopt/scip.pyx [hunk 2]
```diff
@@ -3562,28 +3562,32 @@ cdef class Model:
             raise Warning("method cannot be called before problem is solved")
         return self.getSolObjVal(self._bestSol, original)
 
-    def getSolVal(self, Solution sol, Variable var):
+    def getSolVal(self, Solution sol, Expr expr):
         """Retrieve value of given variable in the given solution or in
```
Comment (mattmilten) [inline src/pyscipopt/scip.pyx]:
```suggestion
        """Retrieve value of given variable or expression in the given solution or in
```

File: src/pyscipopt/scip.pyx [hunk 3]
```diff
@@ -3562,28 +3562,32 @@ cdef class Model:
...
-        return SCIPgetSolVal(self._scip, sol.sol, var.scip_var)
+        if isinstance(expr, Variable):
+            var = <Variable> expr
+            return SCIPgetSolVal(self._scip, sol.sol, var.scip_var)
+        else:
+            return expr._evaluate(sol)
 
-    def getVal(self, Variable var):
+    def getVal(self, Expr expr):
         """Retrieve the value of the best known solution.
```
Comment (mattmilten) [inline src/pyscipopt/scip.pyx]:
```suggestion
        """Retrieve the value of the given variable or expression in the best known solution.
```

File: src/pyscipopt/expr.pxi (hunk 1 above)
Comment (fserra) [reply src/pyscipopt/expr.pxi]: I was gonna suggest to change `value` to `solution` :)

What about using `point`? I guess one evaluates at a `point` to get a `value`

File: src/pyscipopt/scip.pyx [hunk 4]
```diff
@@ -3562,28 +3562,32 @@ cdef class Model:
...
 
-    def getSolVal(self, Solution sol, Variable var):
-        """Retrieve value of given variable in the given solution or in
+    def getSolVal(self, Solution sol, Expr expr):
+        """Retrieve value of given variable or expression in the given solution or in
         the LP/pseudo solution if sol == None
 
         :param Solution sol: solution
-        :param Variable var: variable to query the value of
+        :param Expr expr: polynomial expression to query the value of
```
Comment (fserra) [inline src/pyscipopt/scip.pyx]: I would probably be repetitive here and add something like `Note: a variable is also an expression`

File: src/pyscipopt/expr.pxi (hunk 1 above)
Comment (mattmilten) [reply src/pyscipopt/expr.pxi]: Yes, `point` is better.

**Comment by fserra** [comment]
I have a suggestion. Benny changed the way we ask for solution values of variables: instead of `m.getSolVal(var, sol)` one can use `sol[var]`.

Why not extend this to expressions? It would be really cool to have `sol[var^2]` or stuff like that.

**Comment by mattmilten** [comment]
Yeah, I also thought about this. I still would like to keep the "old-fashioned" methods as well.

**Comment by mattmilten** [comment]
Shall we merge this first and then implement the "shortcut" `sol[x^3]`?

**Comment by fserra** [comment]
done, thanks @mcitoler

**Pull request merged**